File: java/eyes.ufg.java/src/main/java/com/applitools/eyes/visualgrid/model/AndroidDeviceInfo.java
package com.applitools.eyes.visualgrid.model;

import com.applitools.eyes.RectangleSize;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

public class AndroidDeviceInfo {
  @JsonProperty("name")
  private final AndroidDeviceName deviceName;
  private final ScreenOrientation screenOrientation;
  @JsonInclude(JsonInclude.Include.NON_NULL)
  private final DeviceAndroidVersion version;

  @JsonIgnore
  private String vhsType;

  RectangleSize size = null;

  public AndroidDeviceInfo(AndroidDeviceName deviceName) {
    this(deviceName, ScreenOrientation.PORTRAIT);
  }

  public AndroidDeviceInfo(AndroidDeviceName deviceName, ScreenOrientation screenOrientation) {
    this(deviceName, screenOrientation, null);
  }

  public AndroidDeviceInfo(AndroidDeviceName deviceName, DeviceAndroidVersion version) {
    this(deviceName, ScreenOrientation.PORTRAIT, version);
  }

  public AndroidDeviceInfo(AndroidDeviceName deviceName, ScreenOrientation screenOrientation, DeviceAndroidVersion version) {
    this.deviceName = deviceName;
    this.screenOrientation = screenOrientation;
    this.version = version;
  }

  public String getDeviceName() {
    return deviceName.getName();
  }

  public ScreenOrientation getScreenOrientation() {
    return screenOrientation;
  }

  public DeviceAndroidVersion getVersion() {
    return version;
  }

  public String getVhsType() {
    return vhsType;
  }

  public void setVhsType(String vhsType) {
    this.vhsType = vhsType;
  }

  @Override
  public String toString() {
    return "AndroidDecideInfo{" +
        "deviceName=" + deviceName +
        ", screenOrientation=" + screenOrientation +
        ", version=" + version +
        '}';
  }
}

File: dotnet/Eyes.Images.DotNet/VisualGrid/AndroidDeviceInfo.cs
using System;
using System.Runtime.Serialization;
using Applitools.Utils;
using Applitools.Utils.Geometry;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Applitools.VisualGrid
{
    public class AndroidDeviceInfo : IRenderBrowserInfo, IEquatable<AndroidDeviceInfo>
    {
        public AndroidDeviceInfo
        (AndroidDeviceName deviceName,
            ScreenOrientation screenOrientation = ScreenOrientation.Portrait,
            AndroidVersion? androidVersion = null)
        {
            DeviceName = deviceName;
            ScreenOrientation = screenOrientation;
            Version = androidVersion;
        }

        public AndroidDeviceName DeviceName { get; }

        [JsonIgnore]
        public RectangleSize Size { get; set; }

        [JsonIgnore]
        public string SerializedDeviceName
        {
            get
            {
                try { return DeviceName.GetAttribute<EnumMemberAttribute>().Value; }
                catch (Exception) { return DeviceName.ToString(); }
            }
        }

        [JsonConverter(typeof(StringEnumConverter))]
        [JsonProperty("screenOrientation")]
        public ScreenOrientation ScreenOrientation { get; }
        
        [JsonProperty("version")]
        public AndroidVersion? Version { get; }

        public bool Equals(AndroidDeviceInfo other)
        {
            if (other == null) return false;

            return DeviceName == other.DeviceName &&
                   ScreenOrientation == other.ScreenOrientation;
        }

        public override bool Equals(object obj)
        {
            return Equals(obj as AndroidDeviceInfo);
        }

        public override int GetHashCode()
        {
            return ToString().GetHashCode();
        }

        public override string ToString()
        {
            return $"{nameof(AndroidDeviceInfo)} {{{DeviceName} {Version} {ScreenOrientation}}}";
        }
    }
}